src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -36171,62 +36155,115 @@ public void UnconstrainedTypeParameter_Return_03()
     static U F18<T, U>(T t) where U : class, T => (U)t;
     static U F19<T, U>(T t) where U : struct, T => (U)t;
     static U F20<T, U>(T t) where U : T, new() => (U)t;
+    static U F21<T, U>(T t) => (U)(object)t;
```

Comment:
>F21 [](start = 13, length = 3)

I'd expect a W warning for `F21` too (cast to `object`).

Reply:
Support was added when I merged from master.

---
In reply to: [207681475](https://github.com/dotnet/roslyn/pull/28956#discussion_r207681475) [](ancestors = 207681475)